core/src/test/scala/integration/kafka/api/ConsumerTopicCreationTest.scala

```diff
@@ -72,12 +71,24 @@ class ConsumerTopicCreationTest(brokerAutoTopicCreationEnable: JBoolean, consume
   @Test
   def testAutoTopicCreation(): Unit = {
     val consumer = createConsumer()
+    val producer = createProducer()
+    val record = new ProducerRecord(topic_1, 0, s"key".getBytes, s"value".getBytes)
+
     adminClient = AdminClient.create(createConfig())
 
-    consumer.subscribe(util.Arrays.asList(topic))
-    consumer.poll(Duration.ofMillis(100))
+    // create `topic_1` and produce a record to it
+    adminClient.createTopics(Collections.singleton(new NewTopic(topic_1, 1, 1))).all.get
+    producer.send(record).get
+
+    consumer.subscribe(util.Arrays.asList(topic_1, topic_2))
+
+    // Wait until the produced record was consumed. This guarantees that metadata request for `topic_2` was sent to the
+    // broker.
+    TestUtils.waitUntilTrue(() => {
+      consumer.poll(Duration.ofMillis(100)).count > 0
+    }, "Timed out waiting to consume")
 
-    val topicCreated = adminClient.listTopics.names.get.contains(topic)
+    val topicCreated = adminClient.listTopics.names.get.contains(topic_2)
```

Comment:
There may still be a timing issue here. We are relying both on the creation of the topic and propagation of metadata to be completed. Perhaps we could turn this into a `waitUntilTrue` if we expect the topic to exist?

Reply:
Good point. I updated to check ZK for existence of the topic znode. This will also be a better check for the case where we do not expect the topic to be created. Let me know what you think.